Detectors/MUON/MCH/Workflow/src/DataDecoderSpec.cxx

```diff
@@ -158,30 +174,44 @@ class DataDecoderTask
     }
     mDecoder->computeDigitsTime();
 
-    auto& digits = mDecoder->getOutputDigits();
-    auto& orbits = mDecoder->getOrbits();
+    mRofFinder.process(mDecoder->getDigits(), mDummyROFs);
 
-    if (mDebug) {
-      dumpOrbits(orbits);
-      for (auto d : digits) {
-        std::cout << " DE# " << d.getDetID() << " PadId " << d.getPadID() << " ADC " << d.getADC() << " time " << d.getTime() << std::endl;
+    if (mCheckROFs) {
+      if (mRofFinder.isRofTimeMonotonic() == false) {
+        LOG(ERROR) << "WARNING - ROF time does not increase monotonically";
+      }
+      if (mRofFinder.isDigitsTimeAligned() == false) {
+        LOG(ERROR) << "WARNING - some digits have misaligned time stamps";
       }
     }
+
+    //auto& digits = mDecoder->getDigits();
+    auto& orbits = mDecoder->getOrbits();
+
     // send the output buffer via DPL
-    size_t digitsSize, orbitsSize;
-    char* digitsBuffer = createBuffer(digits, digitsSize);
+    size_t digitsSize, rofsSize, orbitsSize;
+    char* digitsBuffer = mRofFinder.storeDigits(digitsSize);
+    char* rofsBuffer = mRofFinder.storeROFRecords(rofsSize);
     char* orbitsBuffer = createBuffer(orbits, orbitsSize);
 
+    if (mDebug) {
+      std::cout << "digitsSize " << digitsSize << "  rofsSize " << rofsSize << "  orbitsSize " << orbitsSize << std::endl;
+    }
+
     // create the output message
     auto freefct = [](void* data, void*) { free(data); };
-    pc.outputs().adoptChunk(Output{"MCH", "DIGITS", 0}, digitsBuffer, digitsSize, freefct, nullptr);
-    pc.outputs().adoptChunk(Output{"MCH", "ORBITS", 0}, orbitsBuffer, orbitsSize, freefct, nullptr);
+    pc.outputs().adoptChunk(Output{header::gDataOriginMCH, "DIGITS", 0}, digitsBuffer, digitsSize, freefct, nullptr);
+    pc.outputs().adoptChunk(Output{header::gDataOriginMCH, "ROFS", 0}, rofsBuffer, rofsSize, freefct, nullptr);
```

Comment:
other workflows expect the digit rofs spec to be named "DIGITROFS" (because there's also CLUSTEROFs, etc...)

Reply:
Ok, I will change the output spec.